crypto/ocsp/ocsp_test.cc

```diff
@@ -218,3 +525,101 @@ TEST(OCSPTest, TestBasic) {
+    ASSERT_TRUE(issuer_candidate);
+    const int issuer_value = X509_check_issued(issuer_candidate, subject);
+    if (issuer_value == X509_V_OK) {
+      issuer = issuer_candidate;
+      break;
+    }
+  }
+  ASSERT_TRUE(issuer);
+
+  // Expect basic verify here, but we skip step for now since functionality has
+  // not been implemented yet.
+
+  int status = 0;
+  int reason = 0;
+  // Convert issuer certificate to |OCSP_CERTID|
+  bssl::UniquePtr<OCSP_CERTID> cert_id = bssl::UniquePtr<OCSP_CERTID>(OCSP_cert_to_id(EVP_sha1(), subject, issuer));
+  ASSERT_TRUE(cert_id);
+
+  ASN1_GENERALIZEDTIME *revtime, *thisupd, *nextupd;
+  // Actual verification of the response
+  const int ocsp_resp_find_status_res = OCSP_resp_find_status(basic_response.get(), cert_id.get(), &status, &reason, &revtime, &thisupd, &nextupd);
+  ASSERT_EQ(1, ocsp_resp_find_status_res);
+  ASSERT_EQ(V_OCSP_CERTSTATUS_GOOD, status);
+}
+
+TEST(OCSPTest, TestRevokedOCSP) {
+  bssl::UniquePtr<OCSP_RESPONSE> ocsp_response;
+  bssl::UniquePtr<OCSP_BASICRESP> basic_response;
+
+  ocsp_response = LoadOCSP_RESPONSE(ocsp_revoked_response_der);
+  ASSERT_TRUE(ocsp_response);
+
+  int ocsp_status = OCSP_response_status(ocsp_response.get());
+  ASSERT_EQ(OCSP_RESPONSE_STATUS_SUCCESSFUL, ocsp_status);
+
+  basic_response = bssl::UniquePtr<OCSP_BASICRESP>(OCSP_response_get1_basic(ocsp_response.get()));
+  ASSERT_TRUE(basic_response);
+
+  // Set up trust store and certificate chain
+  bssl::UniquePtr<X509_STORE> trust_store(X509_STORE_new());
+  X509_STORE_add_cert(trust_store.get(),LoadX509fromPEM(ca_cert).get());
+  bssl::UniquePtr<STACK_OF(X509)> server_cert_chain = CertsToStack(
+      {LoadX509fromPEM(server_cert).get(),LoadX509fromPEM(ca_cert).get()});;
+
+  X509 *subject = sk_X509_value(server_cert_chain.get(), 0);
+  X509 *issuer = nullptr;
+  // Find the issuer in the certificate chain. If it's not there, fail everything.
+  for (size_t i = 0; i < sk_X509_num(server_cert_chain.get()); i++) {
+    X509 *issuer_candidate = sk_X509_value(server_cert_chain.get(), i);
+    ASSERT_TRUE(issuer_candidate);
+    const int issuer_value = X509_check_issued(issuer_candidate, subject);
+    if (issuer_value == X509_V_OK) {
+      issuer = issuer_candidate;
+      break;
+    }
+  }
+  ASSERT_TRUE(issuer);
+
+  // Expect basic verify here, but we skip step for now since functionality has
+  // not been implemented yet.
+
+  int status = 0;
+  int reason = 0;
+  // Convert issuer certificate to |OCSP_CERTID|
+  bssl::UniquePtr<OCSP_CERTID> cert_id = bssl::UniquePtr<OCSP_CERTID>(OCSP_cert_to_id(EVP_sha1(), subject, issuer));
+  ASSERT_TRUE(cert_id);
+
+  ASN1_GENERALIZEDTIME *revtime, *thisupd, *nextupd;
+  // Actual verification of the response
+  const int ocsp_resp_find_status_res = OCSP_resp_find_status(basic_response.get(), cert_id.get(), &status, &reason, &revtime, &thisupd, &nextupd);
+  ASSERT_EQ(1, ocsp_resp_find_status_res);
+  ASSERT_EQ(V_OCSP_CERTSTATUS_REVOKED, status);
+}
```

Comment:
What about the third status code `V_OCSP_CERTSTATUS_UNKNOWN`?

Reply:
s2n has no tests in regards to the third status code `V_OCSP_CERTSTATUS_UNKNOWN`. I've already took note of this case and have added to the list of related tests to add after bare minimum functionality support for s2n is done. I'll provide a document list of tests that we should consider adding.